rest-api/dao/physical_measurements_dao.py

```diff
@@ -46,15 +47,33 @@ def get_with_session(self, session, obj_id, **kwargs):
       ParticipantDao().validate_participant_reference(session, result)
     return result
 
-  def get_with_children(self, physical_measurements_id):
+  def get_with_children(self, physical_measurements_id, for_update=False):
     with self.session() as session:
       query = session.query(PhysicalMeasurements) \
           .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
               Measurement.measurements)) \
           .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
               Measurement.qualifiers))
+
+      if for_update:
+        query = query.with_for_update()
+        return query.get(physical_measurements_id)
+
+      return query.get(physical_measurements_id)
+
+  def get_with_children_with_session(self, session, physical_measurements_id, for_update=False):
+    query = session.query(PhysicalMeasurements) \
+      .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
+        Measurement.measurements)) \
+      .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
+        Measurement.qualifiers))
+
+    if for_update:
+      query = query.with_for_update()
       return query.get(physical_measurements_id)
```

Comment:
Nit: you could remove this line and the method still works.

Reply:
done